Add button docs for Madvr (#33910)
home-assistant/home-assistant.github.io: source/_integrations/madvr.markdown

- `KeyPress (MENU | UP | DOWN | LEFT | RIGHT | OK | INPUT | SETTINGS | RED | GREEN | BLUE | YELLOW | POWER)`
- `KeyHold (MENU | UP | DOWN | LEFT | RIGHT | OK | INPUT | SETTINGS | RED | GREEN | BLUE | YELLOW | POWER)`

> ### Usage Example

Reviewer: ```suggestion
### Usage example
```
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization), also in headings

`remote.send_command`: `KeyPress, MENU`

## Binary sensor

> - `Power state` is On when the device is turned on.
> - `Signal state` is On when the device is receiving a signal from the source.

Reviewer: ```suggestion
- `Power state`: Is `On` when the device is turned on.
- `Signal state`: Is `On` when the device is receiving a signal from the source.
```

These can be used for various purposes, such as triggering your masking system based on the detected aspect ratio.

## Sensor

> ### Enabled By Default

Reviewer: ```suggestion
### Enabled by default
```

- `Outgoing frame rate`: The frame rate of the outgoing video signal.
- `Outgoing resolution`: The resolution of the outgoing video signal.

> ### Disabled By Default

Reviewer: ```suggestion
### Disabled by default
```

- `Masking resolution`: The resolution for the current masking setting.
- `Masking decimal`: The masking ratio as a decimal value.
- `Masking integer`: The masking ratio as an integer ratio.

> ### Button

Reviewer: ```suggestion
### Buttons
```

The integration creates buttons which send commands to the device. You can add these buttons in the UI to build a virtual remote or add them in scripts/automations.

These buttons control navigation and menus:

- `Up`: Simulates pressing the up button.
- `Down`: Simulates pressing the down button.
- `Left`: Simulates pressing the left button.
- `Right`: Simulates pressing the right button.
- `OK`: Simulates pressing the OK button.
- `Back`: Simulates pressing the back button.
- `Red`: Simulates pressing the red button.
- `Green`: Simulates pressing the green button.
- `Blue`: Simulates pressing the blue button.
- `Yellow`: Simulates pressing the yellow button.
- `Magenta`: Simulates pressing the magenta button.
- `Cyan`: Simulates pressing the cyan button.
- `Open info menu`: Opens the info menu.
- `Open settings menu`: Opens the settings menu.
- `Open configuration menu`: Opens the configuration menu.
- `Open profiles menu`: Opens the profiles menu.
- `Open test patterns menu`: Opens the test patterns menu.

> These buttons are for debugging or demoing:

Reviewer: ```suggestion
These buttons are for debugging or demonstrating:
```

- `Outgoing signal type`: The type of the outgoing signal (3D or 2D).

These sensors are disabled because their values are not commonly needed but they can be enabled in the UI according to your needs.

## Buttons

The integration creates buttons which send commands to the device. You can add these buttons in the UI to build a virtual remote or add them in scripts/automations.

> ### Navigation & Menus

Reviewer: ```suggestion
### Navigation & menus
```

- `Up`: Simulates pressing the up button.
- `Down`: Simulates pressing the down button.
- `Left`: Simulates pressing the left button.
- `Right`: Simulates pressing the right button.
- `OK`: Simulates pressing the OK button.
- `Back`: Simulates pressing the back button.
- `Red`: Simulates pressing the red button.
- `Green`: Simulates pressing the green button.
- `Blue`: Simulates pressing the blue button.
- `Yellow`: Simulates pressing the yellow button.
- `Magenta`: Simulates pressing the magenta button.
- `Cyan`: Simulates pressing the cyan button.
- `Open info menu`: Opens the info menu.
- `Open settings menu`: Opens the settings menu.
- `Open configuration menu`: Opens the configuration menu.
- `Open profiles menu`: Opens the profiles menu.
- `Open test patterns menu`: Opens the test patterns menu.

> ### Debugging & Demos

Reviewer: ```suggestion
### Debugging & demos
```

> - `Reset temporary` resets temporary settings to their default values.
> - `Toggle tone map` toggles the tone mapping feature on or off.
> - `Toggle highlight recovery` toggles the highlight recovery feature on or off.
> - `Toggle contrast recovery` toggles the contrast recovery feature on or off.
> - `Toggle shadow recovery` toggles the shadow recovery feature on or off.
> - `Toggle 3D LUT` toggles the 3D Look-Up Table (LUT) feature on or off.
> - `Toggle screen boundaries` toggles the display of screen boundaries.
> - `Toggle histogram` toggles the display of the video histogram.
> - `Toggle debug OSD` toggles the debug on-screen display.
> - `Refresh license info` refreshes the license information.
> - `Force 1080p60 output` forces the output to 1080p at 60Hz (basically safe mode).

Reviewer: ```suggestion
- `Reset temporary`: Resets temporary settings to their default values.
- `Toggle tone map`: Toggles the tone mapping feature on or off.
- `Toggle highlight recovery`: Toggles the highlight recovery feature on or off.
- `Toggle contrast recovery`: Toggles the contrast recovery feature on or off.
- `Toggle shadow recovery`: Toggles the shadow recovery feature on or off.
- `Toggle 3D LUT`: Toggles the 3D Look-Up Table (LUT) feature on or off.
- `Toggle screen boundaries`: Toggles the display of screen boundaries.
- `Toggle histogram`: Toggles the display of the video histogram.
- `Toggle debug OSD`: Toggles the debug on-screen display.
- `Refresh license info`: Refreshes the license information.
- `Force 1080p60 output`: Forces the output to 1080p at 60Hz (basically safe mode).
```

> - `Force 1080p60 output` forces the output to 1080p at 60Hz (basically safe mode).

Reviewer: > **Fix loose punctuation marks.**
> 
> Remove the loose punctuation marks before each command to improve readability.
> 
> ```diff
> - - `Up`: Simulates pressing the up button.
> - - `Down`: Simulates pressing the down button.
> - - `Left`: Simulates pressing the left button.
> - - `Right`: Simulates pressing the right button.
> - - `OK`: Simulates pressing the OK button.
> - - `Back`: Simulates pressing the back button.
> - - `Red`: Simulates pressing the red button.
> - - `Green`: Simulates pressing the green button.
> - - `Blue`: Simulates pressing the blue button.
> - - `Yellow`: Simulates pressing the yellow button.
> - - `Magenta`: Simulates pressing the magenta button.
> - - `Cyan`: Simulates pressing the cyan button.
> - - `Open info menu`: Opens the info menu.
> - - `Open settings menu`: Opens the settings menu.
> - - `Open configuration menu`: Opens the configuration menu.
> - - `Open profiles menu`: Opens the profiles menu.
> - - `Open test patterns menu`: Opens the test patterns menu.
> - - `Reset temporary`: Resets temporary settings to their default values.
> - - `Toggle tone map`: Toggles the tone mapping feature on or off.
> - - `Toggle highlight recovery`: Toggles the highlight recovery feature on or off.
> - - `Toggle contrast recovery`: Toggles the contrast recovery feature on or off.
> - - `Toggle shadow recovery`: Toggles the shadow recovery feature on or off.
> - - `Toggle 3D LUT`: Toggles the 3D Look-Up Table (LUT) feature on or off.
> - - `Toggle screen boundaries`: Toggles the display of screen boundaries.
> - - `Toggle histogram`: Toggles the display of the video histogram.
> - - `Toggle debug OSD`: Toggles the debug on-screen display.
> - - `Refresh license info`: Refreshes the license information.
> - - `Force 1080p60 output`: Forces the output to 1080p at 60Hz (basically safe mode).
> + - `Up` simulates pressing the up button.
> + - `Down` simulates pressing the down button.
> + - `Left` simulates pressing the left button.
> + - `Right` simulates pressing the right button.
> + - `OK` simulates pressing the OK button.
> + - `Back` simulates pressing the back button.
> + - `Red` simulates pressing the red button.
> + - `Green` simulates pressing the green button.
> + - `Blue` simulates pressing the blue button.
> + - `Yellow` simulates pressing the yellow button.
> + - `Magenta` simulates pressing the magenta button.
> + - `Cyan` simulates pressing the cyan button.
> + - `Open info menu` opens the info menu.
> + - `Open settings menu` opens the settings menu.
> + - `Open configuration menu` opens the configuration menu.
> + - `Open profiles menu` opens the profiles menu.
> + - `Open test patterns menu` opens the test patterns menu.
> + - `Reset temporary` resets temporary settings to their default values.
> + - `Toggle tone map` toggles the tone mapping feature on or off.
> + - `Toggle highlight recovery` toggles the highlight recovery feature on or off.
> + - `Toggle contrast recovery` toggles the contrast recovery feature on or off.
> + - `Toggle shadow recovery` toggles the shadow recovery feature on or off.
> + - `Toggle 3D LUT` toggles the 3D Look-Up Table (LUT) feature on or off.
> + - `Toggle screen boundaries` toggles the display of screen boundaries.
> + - `Toggle histogram` toggles the display of the video histogram.
> + - `Toggle debug OSD` toggles the debug on-screen display.
> + - `Refresh license info` refreshes the license information.
> + - `Force 1080p60 output` forces the output to 1080p at 60Hz (basically safe mode).
> ```
> 
> Committable suggestion
> > ‼️ **IMPORTANT**
> > Carefully review the code before committing. Ensure that it accurately replaces the highlighted code, contains no missing lines, and has no issues with indentation. Thoroughly test & benchmark the code to ensure it meets the requirements.
> 
> Tools

why would this improve readability?

> - `Force 1080p60 output` forces the output to 1080p at 60Hz (basically safe mode).

Reviewer: > **Fix loose punctuation marks.**
> 
> Remove the loose punctuation marks before each command to improve readability.
> 
> ```diff
> - - `Up`: Simulates pressing the up button.
> - - `Down`: Simulates pressing the down button.
> - - `Left`: Simulates pressing the left button.
> - - `Right`: Simulates pressing the right button.
> - - `OK`: Simulates pressing the OK button.
> - - `Back`: Simulates pressing the back button.
> - - `Red`: Simulates pressing the red button.
> - - `Green`: Simulates pressing the green button.
> - - `Blue`: Simulates pressing the blue button.
> - - `Yellow`: Simulates pressing the yellow button.
> - - `Magenta`: Simulates pressing the magenta button.
> - - `Cyan`: Simulates pressing the cyan button.
> - - `Open info menu`: Opens the info menu.
> - - `Open settings menu`: Opens the settings menu.
> - - `Open configuration menu`: Opens the configuration menu.
> - - `Open profiles menu`: Opens the profiles menu.
> - - `Open test patterns menu`: Opens the test patterns menu.
> - - `Reset temporary`: Resets temporary settings to their default values.
> - - `Toggle tone map`: Toggles the tone mapping feature on or off.
> - - `Toggle highlight recovery`: Toggles the highlight recovery feature on or off.
> - - `Toggle contrast recovery`: Toggles the contrast recovery feature on or off.
> - - `Toggle shadow recovery`: Toggles the shadow recovery feature on or off.
> - - `Toggle 3D LUT`: Toggles the 3D Look-Up Table (LUT) feature on or off.
> - - `Toggle screen boundaries`: Toggles the display of screen boundaries.
> - - `Toggle histogram`: Toggles the display of the video histogram.
> - - `Toggle debug OSD`: Toggles the debug on-screen display.
> - - `Refresh license info`: Refreshes the license information.
> - - `Force 1080p60 output`: Forces the output to 1080p at 60Hz (basically safe mode).
> + - `Up` simulates pressing the up button.
> + - `Down` simulates pressing the down button.
> + - `Left` simulates pressing the left button.
> + - `Right` simulates pressing the right button.
> + - `OK` simulates pressing the OK button.
> + - `Back` simulates pressing the back button.
> + - `Red` simulates pressing the red button.
> + - `Green` simulates pressing the green button.
> + - `Blue` simulates pressing the blue button.
> + - `Yellow` simulates pressing the yellow button.
> + - `Magenta` simulates pressing the magenta button.
> + - `Cyan` simulates pressing the cyan button.
> + - `Open info menu` opens the info menu.
> + - `Open settings menu` opens the settings menu.
> + - `Open configuration menu` opens the configuration menu.
> + - `Open profiles menu` opens the profiles menu.
> + - `Open test patterns menu` opens the test patterns menu.
> + - `Reset temporary` resets temporary settings to their default values.
> + - `Toggle tone map` toggles the tone mapping feature on or off.
> + - `Toggle highlight recovery` toggles the highlight recovery feature on or off.
> + - `Toggle contrast recovery` toggles the contrast recovery feature on or off.
> + - `Toggle shadow recovery` toggles the shadow recovery feature on or off.
> + - `Toggle 3D LUT` toggles the 3D Look-Up Table (LUT) feature on or off.
> + - `Toggle screen boundaries` toggles the display of screen boundaries.
> + - `Toggle histogram` toggles the display of the video histogram.
> + - `Toggle debug OSD` toggles the debug on-screen display.
> + - `Refresh license info` refreshes the license information.
> + - `Force 1080p60 output` forces the output to 1080p at 60Hz (basically safe mode).
> ```
> 
> Committable suggestion
> Tools

I disagree. The colon is applied consistently. Also, it sets the term to be explained apart from the description, which makes it easier to scan quickly.

> - `Reset temporary`: Resets temporary settings to their default values.
> - `Toggle tone map`: Toggles the tone mapping feature on or off.
> - `Toggle highlight recovery`: Toggles the highlight recovery feature on or off.
> - `Toggle contrast recovery`: Toggles the contrast recovery feature on or off.
> - `Toggle shadow recovery`: Toggles the shadow recovery feature on or off.
> - `Toggle 3D LUT`: Toggles the 3D Look-Up Table (LUT) feature on or off.
> - `Toggle screen boundaries`: Toggles the display of screen boundaries.
> - `Toggle histogram`: Toggles the display of the video histogram.
> - `Toggle debug OSD`: Toggles the debug on-screen display.
> - `Refresh license info`: Refreshes the license information.
> - `Force 1080p60 output`: Forces the output to 1080p at 60Hz (basically safe mode).

Reviewer: ```suggestion
- `Reset temporary`: Resets temporary settings to their default values.
- `Toggle tone map`: Toggles the tone mapping feature on or off.
- `Toggle highlight recovery`: Toggles the highlight recovery feature on or off.
- `Toggle contrast recovery`: Toggles the contrast recovery feature on or off.
- `Toggle shadow recovery`: Toggles the shadow recovery feature on or off.
- `Toggle 3D LUT`: Toggles the 3D Look-Up Table (LUT) feature on or off.
- `Toggle screen boundaries`: Toggles the display of screen boundaries.
- `Toggle histogram`: Toggles the display of the video histogram.
- `Toggle debug OSD`: Toggles the debug on-screen display.
- `Refresh license info`: Refreshes the license information.
- `Force 1080p60 output`: Forces the output to 1080p at 60&nbsp;Hz (basically safe mode).
```
